packages/ast/src/index.js

```diff
@@ -33,6 +33,32 @@ export function signature(object: string, name: string): SignatureMap {
   return sign[0];
 }
 
+export function functionNameMetadata(
+  value: string,
+  index: number
+): FunctionNameMetadata {
+  return {
+    type: "FunctionNameMetadata",
+    value,
+    index
+  };
+}
+
+export function moduleMetadata(
+  sections: Array<SectionMetadata>,
+  functionNames: Array<FunctionNameMetadata>
+): ModuleMetadata {
+  const n: ModuleMetadata = {
+    type: "ModuleMetadata",
+    sections
+  };
+
+  if (functionNames.length) {
+    n.functionNames = functionNames;
+  }
+  return n;
+}
```

Comment:
I'm not sure the metadata should be a node in our AST? I already model it here https://github.com/xtuc/webassemblyjs/blob/7b3e82af4bf89b49af46cac5d64e8659a06b8037/packages/ast/src/index.js#L93 as a property on the module, what do you think?

Reply:
I've kept the metadata in the same place. The change I've made is to give it a `type` property so that it can be traversed by the AST parser. Does that sound OK?